### PR TITLE
CI: windows cp311 use-pythran=false full, sdist GHA

### PR DESCRIPTION
**Pull request by andyfaff**
Completes point 3 of https://github.com/scipy/scipy/issues/15814#issuecomment-1515076563

i.e. Windows/cp311/use-pythran=false using `python -m build` to make the sdist and build the wheel from that.

Please check that I'm running the full test suite correctly.

### CODE REVIEW

**Comment by andyfaff** [comment]
I wonder if the failures in https://github.com/scipy/scipy/actions/runs/4761936327/jobs/8463649690?pr=18333 are down to the same issue that required the extra `meson install` step in #18325

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -123,3 +123,58 @@ jobs:
...
+
+          $env:wheel_name=Get-ChildItem -Path dist/* -Include *.whl
+          delvewheel repair --add-path c:\opt\openblas\openblas_dll -w dist $env:wheel_name
+          
+          python -m pip install --only-binary=scipy --no-index --find-links=dist scipy
+
+      - name: Test
+        run: |
+          cd $RUNNER_TEMP
+          pytest -k test --pyargs scipy
```
Comment (rgommers) [inline .github/workflows/windows.yml:179]: `-k test` seems unnecessary, it's a regex that selects every single test. Running the full test suite like this works; you may want to append a `  # runs full test suite` comment (`pytest --pyargs scipy` has the opposite default as `dev.py` and `scipy.test()`, which may not be obvious to everyone).

**Review by rgommers — APPROVED** [review]
Looks great, thanks @andyfaff. I only have some minor comments, looks good to merge other than that, CI log shows this works as expected.

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -123,3 +123,58 @@ jobs:
...
+      - name: Install OpenBLAS
+        shell: bash
+        run: |
+          set -xe
+          bash tools/wheels/cibw_before_build_win.sh .
+          echo "PKG_CONFIG_PATH=c:\opt\64\lib\pkgconfig;" >> $GITHUB_ENV
+
+      - name: pip-packages
+        run: |
+          python -m pip install build delvewheel numpy cython pybind11 meson-python meson ninja pytest pytest-xdist pytest-timeout pooch rich_click click doit pydevtool
```
Comment (rgommers) [inline .github/workflows/windows.yml:165]: one more comment: `dev.py` isn't used in this job, so you can remove the last four packages from this line

File: .github/workflows/windows.yml [hunk 3]
```diff
@@ -123,3 +123,58 @@ jobs:
...
+
+      - name: Build
+        run: |
+          python -m build -Csetup-args="-Duse-pythran=false"
+
+          $env:wheel_name=Get-ChildItem -Path dist/* -Include *.whl
+          delvewheel repair --add-path c:\opt\openblas\openblas_dll -w dist $env:wheel_name
+          
+          python -m pip install --only-binary=scipy --no-index --find-links=dist scipy
```
[the remark this answers is not in the document]
Comment (andyfaff) [reply .github/workflows/windows.yml:174]: Having said that I do work out the exact filename for delvewheel on the previous step, so I can use that.

With delvewheel you can place the output wheel anywhere, doesn't need to overwrite.

[the remark this answers is not in the document]
Comment (rgommers) [reply .github/workflows/windows.yml]: Ah, too long not having touched a Windows machine. I thought it had wildcards, but guess not. Sorry for the noise.

**Review by rgommers — APPROVED** [review]
Comments were addressed, and this looks very clean now. The one CI failure is unrelated, so in it goes. Thanks @andyfaff

**Pull request merged**